utilities/test_suite/HOST/Tensor_host.cpp

```diff
@@ -598,6 +602,84 @@ int main(int argc, char **argv)
 
                     break;
                 }
+                case 82:
+                {
+                    testCaseName = "ricap";
+
+                    int iX = maxWidth;
+                    int iY = maxHeight;
+
+                    Rpp32u initialPermuteArray[batchSize], permutedArray[batchSize * 4], permutationTensor[batchSize * 4];
+                    double randFromDist, randFromDist1;
+                    RpptROI roiPtrInputCropRegion[4];
+                    for (uint i = 0; i < batchSize; i++)
+                    {
+                        initialPermuteArray[i] = i;
+                    }
+
+                    if(qaFlag)
+                    {
+                        for(int i=0;i<4;i++)
+                        memcpy(permutedArray + (batchSize * i), initialPermuteArray, batchSize * sizeof(Rpp32u));
+                    }
+                    else
+                    {
+                        float betaParam = 0.3;
+                        std::random_device rd;
+                        std::mt19937 gen(rd());
+                        static std::uniform_real_distribution<double> unif(0.3, 0.7);
+                        double p = unif(gen);
+                        randFromDist = boost::math::ibeta_inv(betaParam, betaParam, p);
+
+                        std::random_device rd1;
+                        std::mt19937 gen1(rd1());
+                        static std::uniform_real_distribution<double> unif1(0.3, 0.7);
+                        double p1 = unif1(gen1);
+                        randFromDist1 = boost::math::ibeta_inv(betaParam, betaParam, p1);
+
+                        for(int i=0;i<4;i++)
+                        {
+                            randomize(initialPermuteArray, batchSize);
+                            memcpy(permutedArray + (batchSize * i), initialPermuteArray, batchSize * sizeof(Rpp32u));
+                        }
+                    }
+
+                    for (uint i = 0, j = 0; j < batchSize * 4; i++, j += 4)
```

Comment:
Please create a generic function in the test_suite common .h file we have. Name it intuitively "_1D_4channel_pln_to_pkd_..()
Pass relevant arguments and call modularized code from host and hip test suites. Now please combine all the if() to one.

Also, for this function, suggesst to add inline comments everywhere - on why we have the mt19937, std::unif, boost::math::ibeta_inv, randomize(), and std::round before we set the ROI, to increase readability for a 3rd party since its not intuitive

Reply:
Done